lib/pkgcloud/openstack/cdn/client/services.js

```diff
@@ -0,0 +1,311 @@
+ * @param {String}          [details.origins[n].rules[n].name]        the name of the rule
+ * @param {String}          [details.origins[n].rules[n].request_url] the request URL this rule should match (regex supported)
+ * @param {Array}           [details.caching]                         list of TTL rules for assets of this service
+ * @param {Object}          [details.caching[n]]                      information about a TTL rule
+ * @param {String}          [details.caching[n].name]                 the name of the TTL rule
+ * @param {Number}          [details.caching[n].ttl]                  the TTL value, in seconds?
+ * @param {Array}           [details.caching[n].rules]                list of rules that determine if this TTL should be applied to an asset
+ * @param {Object}          [details.caching[n].rules[n]]             information about a TTL rule
+ * @param {String}          [details.caching[n].rules[n].name]        the name of the TTL rule 
+ * @param {String}          [details.caching[n].rules[n].request_url] the request URL this rule should match (regex supported)
+ * @param {Array}           [details.restrictions]                    list of restrictions on who can access new service
+ * @param {Object}          [details.restrictions[n]]                 information about an access restriction
+ * @param {String}          [details.restrictions[n].name]            the name of the restriction
+ * @param {Array}           [details.restrictions[n].rules]           list of restrition rules
+ * @param {Object}          [details.restrictions[n].rules[n]]        information about a restriction rule
+ * @param {String}          [details.restrictions[n].rules[n].name]   the name of the restriction rule
+ * @param {String}          [details.restrictions[n].rules[n].referrer] the domain from which the new service can be accessed
+ * @param {String}          details.flavorId                          the ID of the flavor to use for this service
+ * @param callback
+ * @returns {request|null}
+ */
+exports.createService = function (details, callback) {
+  if (typeof details === 'function') {
+    callback = details;
+    details = {};
+  }
+
+  details = details || {};
+
+  if (!validateProperties(['name', 'domains', 'origins', 'flavorId'], details,
+    'options.%s is a required argument.', callback)) {
+    return;
+  }
+
+  var self = this,
+    createOptions = {
+      method: 'POST',
+      path: _urlPrefix,
+      body: {
+        name: details.name,
+        domains: details.domains,
+        origins: details.origins,
+        flavor_id: details.flavorId
+      }
+    };
+
+  if (details.caching) {
+    createOptions.body.caching = details.caching;
+  }
+
+  if (details.restrictions) {
+    createOptions.body.restrictions = details.restrictions;
+  }
+
+  return self._request(createOptions, function (err, body, res) {
+    if (err) {
+      return callback(err);
```

Comment:
Opinion seems to be divided on this matter:

```
$ grep -inr 'callback\ *(erro\?r\?)' lib/ | wc -l
     356

$  grep -inr 'return\ \+callback\ *(erro\?r\?)' lib/ | wc -l
     140
```

Just `callback(err)` occurs about 1.5 times as `return callback(err)`. Which one do you want me to follow? I'll update all references in this PR to that style so they are consistent within the CDN service at least.